packages/components/nodes/multimodal/OpenAI/VLLMChain.ts

```diff
@@ -0,0 +1,201 @@
+            top_p: this.topP,
+            messages: []
+        }
+        if (this.maxTokens) vRequest.max_tokens = this.maxTokens
+        else vRequest.max_tokens = 1024
+
+        const userRole: any = { role: 'user' }
+        userRole.content = []
+        userRole.content.push({
+            type: 'text',
+            text: userInput
+        })
+        if (this.whisperConfig && this.imageUrls && this.imageUrls.length > 0) {
+            const audioUploads = this.getAudioUploads(this.imageUrls)
+            for (const url of audioUploads) {
+                const filePath = path.join(getUserHome(), '.flowise', 'gptvision', url.data, url.name)
+
+                // as the image is stored in the server, read the file and convert it to base64
+                const audio_file = fs.createReadStream(filePath)
+                if (this.whisperConfig.purpose === 'transcription') {
+                    const transcription = await this.client.audio.transcriptions.create({
+                        file: audio_file,
+                        model: 'whisper-1'
+                    })
+                    userRole.content.push({
+                        type: 'text',
+                        text: transcription.text
+                    })
+                } else if (this.whisperConfig.purpose === 'translation') {
+                    const translation = await this.client.audio.translations.create({
+                        file: audio_file,
+                        model: 'whisper-1'
+                    })
+                    userRole.content.push({
+                        type: 'text',
+                        text: translation.text
+                    })
+                }
+            }
+        }
+        if (this.imageUrls && this.imageUrls.length > 0) {
+            const imageUploads = this.getImageUploads(this.imageUrls)
+            for (const url of imageUploads) {
+                let bf = url.data
+                if (url.type == 'stored-file') {
+                    const filePath = path.join(getUserHome(), '.flowise', 'gptvision', url.data, url.name)
+
+                    // as the image is stored in the server, read the file and convert it to base64
+                    const contents = fs.readFileSync(filePath)
+                    bf = 'data:' + url.mime + ';base64,' + contents.toString('base64')
+                }
+                userRole.content.push({
+                    type: 'image_url',
+                    image_url: {
+                        url: bf,
+                        detail: this.imageResolution
+                    }
+                })
+            }
+        }
+        vRequest.messages.push(userRole)
+        if (this.prompt && this.prompt instanceof ChatPromptTemplate) {
+            let chatPrompt = this.prompt as ChatPromptTemplate
+            chatPrompt.promptMessages.forEach((message: any) => {
+                if (message instanceof SystemMessagePromptTemplate) {
```

Comment:
should we also check instanceof `HumanMessagePromptTemplate`?

Reply:
Yes, and pushed changes.